File: libs/services/src/main/java/net/medvedmike/vsu/demo/service/UserServiceImpl.java
package net.medvedmike.vsu.demo.service;

import net.medvedmike.vsu.demo.annotation.system.audit.AuditBegin;
import net.medvedmike.vsu.demo.annotation.system.audit.AuditComplete;
import net.medvedmike.vsu.demo.annotation.system.audit.AuditError;
import net.medvedmike.vsu.demo.domain.User;
import net.medvedmike.vsu.demo.dto.UserDTO;
import net.medvedmike.vsu.demo.domain.UserView;
import net.medvedmike.vsu.demo.repository.UserRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.*;

@Service
public class UserServiceImpl extends BaseService implements UserService {

	@Autowired
	private UserRepository userRepository;

	@AuditError
	@Override
	public List<UserView> getUsers(String login, String name, String surname, String email, Date birthday,
								   int page, int count, String sidx, String sord) {
		return userRepository.getUserViews(login, name, surname, email, birthday, page, count, sidx, sord);
	}

	@AuditError
	@Override
	public UserDTO getDemo(Long id) {
		User user = userRepository.find(id);
		return new UserDTO(user);
	}

	@AuditBegin
	@AuditComplete
	@AuditError
	@Override
	public void createDemo(UserDTO dto) {
		User model = dto.toModel();
		userRepository.create(model);
		dto.setId(model.getId());
	}

	@AuditBegin
	@AuditComplete
	@AuditError
	@Override
	public void updateDemo(UserDTO dto) {
		User model = dto.toModel();
		userRepository.update(model);
	}

	@AuditBegin
	@AuditComplete
	@AuditError
	@Override
	public void deleteDemo(Long ... id) {
		userRepository.delete(Arrays.asList(id));
	}

}

File: libs/services/src/main/kotlin/ru/michaelilyin/blog/services/administrative/UserServiceImpl.kt
package ru.michaelilyin.blog.services.administrative

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service
import ru.michaelilyin.blog.annotation.audit.AuditBegin
import ru.michaelilyin.blog.annotation.audit.AuditComplete
import ru.michaelilyin.blog.annotation.audit.AuditError
import ru.michaelilyin.blog.dto.administrative.User
import ru.michaelilyin.blog.repository.administrative.UserRepository
import ru.michaelilyin.blog.services.BaseService
import java.util.*

/**
 * TODO: javadoc
 * Created by Michael Ilyin on 15.11.2016.
 */
@Service
open class UserServiceImpl @Autowired() constructor(
        private val userRepository: UserRepository
): BaseService(), UserService {

    @AuditError
    override fun getUsers(page: Int, count: Int): List<User> {
        val users = userRepository.getUsers(page, count)
        return users.map(::User)
    }

    @AuditError
    override fun findUser(id: Long): Optional<User> {
        val user = userRepository.findUser(id)
        return user.map({User(it)})
    }

    @AuditBegin
    @AuditComplete
    @AuditError
    override fun createUser(user: User): Long {
        val model = user.model()
        return userRepository.createUser(model)
    }
}